pkg/nsx/services/common/store.go

```diff
@@ -1,61 +1,32 @@
 package common
 
 import (
-	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
 
 	vapierrors "github.com/vmware/vsphere-automation-sdk-go/lib/vapi/std/errors"
+
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
 	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
-	"k8s.io/client-go/tools/cache"
 
 	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
 )
 
-// IndexFunc is wrapper of cache.IndexFunc, used by cache.Indexers
-func IndexFunc(scope string) cache.IndexFunc {
-	return func(obj interface{}) ([]string, error) {
-		return CRUIDScopeIndexFunc(scope, obj)
-	}
-}
-
-// CRUIDScopeIndexFunc returns the index of the object.
-// The index is the value of the tag with scope CRUID, the unique in
-// time and space value for this object at k8s side, which is generated by k8s.
-func CRUIDScopeIndexFunc(scope string, obj interface{}) ([]string, error) {
-	var tags []model.Tag
-	switch v := obj.(type) {
-	case model.SecurityPolicy:
-		tags = v.Tags
-	case model.Group:
-		tags = v.Tags
-	case model.Rule:
-		tags = v.Tags
-	default:
-		return nil, errors.New("invalid object type")
-	}
-	return filterTag(scope, tags), nil
-}
+const (
+	PageSize int64 = 1000
+)
 
-func filterTag(scope string, v []model.Tag) []string {
-	res := make([]string, 0, 5)
-	for _, tag := range v {
-		if *tag.Scope == scope {
-			res = append(res, *tag.Tag)
-		}
-	}
-	return res
-}
-
-// KeyFunc returns the key of the object.
-func KeyFunc(obj interface{}) (string, error) {
-	switch v := obj.(type) {
-	case model.Group:
-		return *v.Id, nil
-	case model.SecurityPolicy:
-		return *v.Id, nil
-	case model.Rule:
-		return *v.Id, nil
-	}
-	return "", nil
+// OperateStore is the interface for store, it should be implemented by subclass
+// TransResourceToStore is the method to transform the resource of type data.StructValue
+// to specific nsx-t side resource and then add it to the store.
+// CRUDResource is the method to create, update and delete the resource to the store based
+// on its tag MarkedForDelete.
+type OperateStore interface {
```

Comment:
How about call it Store or ObjectStore?  Operate is a verb

Reply:
OK